Add documentation for Entities and Components in space_shooter_rs

tigleym opened the pull request:
For #83  

This PR adds a section describing the `Spaceship`, `Enemy`, and `Item` entities and their attached components.

cdsupina commented on book/src/entities-components.md:
```diff
@@ -0,0 +1,74 @@
+of an entity into smaller pieces of data will make it much easier to expand upon
+as the game grows.
+
+We will look at three entities present in space_shooter_rs: `Spaceship`
```
I think that referring to components without the word "Component" at the end (as in `ItemComponent`) is fine. However, we should maybe make a note in this book that our naming convention for our custom components is to have "Component" at the end. I also think that now may be the time to make all of the other components in the game, such as `Spaceship` and `Enemy` be consistent with the rest of the new components that we added and change them to `SpaceshipComponent` and `EnemyComponent`. I think that staying consistent is really important to helping new people learn the code.

cdsupina commented on book/src/entities-components.md:
````diff
@@ -0,0 +1,74 @@
+has properties that are specific to an `Item` entity:
+
+```rust
+pub struct Item {
````
```suggestion
pub struct ItemComponent {
```
I changed this in #87.

cdsupina commented on book/src/entities-components.md:
```diff
@@ -0,0 +1,74 @@
+|SpriteRender|Is rendered to screen|Item, Enemy, Spaceship|
+|Transform|Has a position, scale, and rotation in the game|Item, Enemy, Spaceship|
+
+Again this seems like a lot of components for only three entities. But if we keep
```
```suggestion
Again, this seems like a lot of components for only three entities. But if we keep
```

cdsupina reviewed (changes requested):
Left some suggestions on grammar. More importantly, I left a comment on the consistency of using the "-Component" suffix at the end of the component names. Let me know what you think.

tigleym commented:
Thanks @cdsupina for the review! I've addressed your suggestions:

- Made sure all components had "Component" at the end of their name
- Added some pictures for the architecture.md file
- Added extra content on naming conventions for Custom and Tag components

cdsupina reviewed (approved):
Looks great! Go ahead and merge! I love the inclusion of the graphics from our talk at RustFest!
